curated_transformers/repository/repository.py

```diff
@@ -0,0 +1,279 @@
+import json
+from abc import ABC, abstractmethod
+from json import JSONDecodeError
+from typing import Any, Dict, List, Optional, Tuple
+
+from .._compat import has_safetensors
+from ._hf import (
+    _MODEL_CHECKPOINT_TYPE,
+    HF_MODEL_CONFIG,
+    HF_TOKENIZER_CONFIG,
+    PRIMARY_CHECKPOINT_FILENAMES,
+    SHARDED_CHECKPOINT_INDEX_FILENAMES,
+    SHARDED_CHECKPOINT_INDEX_WEIGHTS_KEY,
+    SPECIAL_TOKENS_MAP,
+    TOKENIZER_JSON,
+    ModelCheckpointType,
+)
+from .file import RepositoryFile
+
+
+class Repository(ABC):
+    """
+    A repository that contains a model or tokenizer.
+    """
+
+    @abstractmethod
+    def file(self, path: str) -> RepositoryFile:
+        """
+        Get a repository file.
+
+        :param path:
+            The path of the file within the repository.
+        :returns:
+            The file.
+        :raises FileNotFoundError:
```

Comment:
Missing `OSError` annotation (which gets raised in subclasses).

Reply:
Added.